sympy/polys/matrices/domainmatrix.py

```diff
@@ -30,6 +30,28 @@ def __init__(self, rows, shape, domain):
         self.shape = shape
         self.domain = domain
 
+    def __getitem__(self, key):
+        i, j = key
+        m, n = self.shape
+
+        if not (isinstance(i, slice) or isinstance(j, slice)):
+            if i >= m:
+                raise IndexError("row index out of range")
+            if j >= n:
+                raise IndexError("column index out of range")
+
+            return DomainScalar(self.rep.getitem(i, j), self.domain)
```

Comment:
I don't think we need to check and raise `IndexError`. That should be the job of `getitem`.

Reply:
Sure will change, I added it here so that if we call `getitem`, then it will be faster if no checking would be there.
Thanks.